ls1intum/Artemis
# `Development`: Migrate legacy course-wide channels

### Checklist
#### General



### Motivation and Context


Originally, the course-default, lecture, and exercise channels were created with is_course_wide = false (since the column didn't exist in the database). Now, for corresponding existing database entries, this field can be set to true.

### Description

A migration script handles the update for course-default, lecture and exercise channels. Lecture and exercise channels are recognized by the existence of an id in lecture_id or exercise_id. Course-default channels are recognized by the absence of both a creator and an associated tutorial group.

### Steps for Testing

Prerequisites:
- 1 Instructor
- One course that has tutorial group channels and also legacy default, lecture and exercise channels (you can recognize these legacy channels by the fact that you can add members to them as an instructor)

1. After deploying, the channels have been migrated, if you cannot add members to them anymore (and all course members are a member automatically)
2. Confirm that only default, lecture, and exercise channels have been migrated

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage






Same as develop

### Screenshots


No UI changes
